core/tests/snapshots.rs

```diff
@@ -71,7 +71,9 @@ struct SnapshotTestConfig {
     full_snapshot_archives_dir: TempDir,
     bank_snapshots_dir: TempDir,
     accounts_dir: PathBuf,
-    _accounts_tmp_dir: TempDir,
+    // keep TempDir::drop from running to retain that dir for the duration of test
+    #[allow(dead_code)]
+    accounts_tmp_dir: TempDir,
```

Comment:
i think this comment properly communicates the implicit intention explicitly, which is added https://github.com/solana-labs/solana/pull/29942#discussion_r1174834165

Reply:
**minor**

Using underscored names to keep references to objects that need to live for some time is one of the useful use cases for underscored names.
So, if this object is needed to keep the `TempDir` around, using an underscore name seems exactly right.
A comment is also helpful, of course.
I think `#[allow(dead_code)]` is a less common choice for this.
So I suggest reverting to the underscored name, but keeping the comment.
```suggestion
    // Keep TempDir::drop from running to retain that dir for the duration of test.
    _accounts_tmp_dir: TempDir,
```

A linter should then catch if someone will actually start using this field.
Which would not happen with `#[allow(dead_code)]`.

Reply:
I should have checked that move is considered "use".
Thank you for the explanation.
I think what you originally did is then perfectly reasonable.

Thinking about it for a few minutes, this seems reasonable as well:
```rust
    _accounts_tmp_dir: accounts_tmp_dir,
```
If it keeps clippy happy.  It can be read as that `accounts_tmp_dir` is used in this function, but it is then not accessed at all, except for the Drop call.

Reply:
> Thinking about it for a few minutes, this seems reasonable as well:
> 
> ```rust
>     _accounts_tmp_dir: accounts_tmp_dir,
> ```

oh, that's good to hear from you. accordingly, I've changed my mind: 6347179921d513aee2664762f494496a1d7b0a5f

I really dislikes `allow(...)`. :)